enterprise/backend/src/metabase_enterprise/harbormaster/client.clj

```diff
@@ -0,0 +1,98 @@
+(ns metabase-enterprise.harbormaster.client
+  "API client for interfacing with Harbormaster on store-api-url."
+  (:require
+   [clj-http.client :as http]
+   [clojure.string :as str]
+   [medley.core :as m]
+   [metabase.models.setting :as setting]
+   [metabase.util.i18n :refer [tru]]
+   [metabase.util.json :as json]
+   [metabase.util.log :as log]
+   [metabase.util.malli :as mu]
+   [metabase.util.malli.registry :as mr]))
+
+(defn- +slash-prefix [s]
+  (cond->> s
+    (not (str/starts-with? s "/")) (str "/")))
+
+(mu/defn- ->requestor [method]
+  (case method
+    :get     http/get
+    :head    http/head
+    :post    http/post
+    :put     http/put
+    :delete  http/delete
+    :options http/options
+    :copy    http/copy
+    :move    http/move
+    :patch   http/patch))
+
+(mu/defn- get-safe-status
+  [response]
+  (when (number? (:status response))
+    (http/success? response)))
+
+(defn- ->config
+  "Returns the config needed to call [[make-request]].
+
+  `->config` either gets the store-api-url and api-key from settings or throws an exception when either are unset or
+  blank."
+  []
+  (let [store-api-url (setting/get-value-of-type :string :store-api-url)
```

Comment:
any reason to use `get-value-of-type` and not just `(store-api-url)`? same with the other setting

Reply:
requiring store-api-url is a module violation, This only skirts the issue. I will require them directly and update the module linter config.